### DISCUSSION TITLE
Does AML read frames from N2P2 datasets?

### DISCUSSION THREAD

**Discussion by lizhiqiang778** [MarsalekGroup/aml#31] [Q&A] [answered]
Hi, developers,
Thank you for your excellent package. I am learning it, and I wonder that AML could read frames from N2P2 datasets, not just CP2K trajectories. I found the n2p2.py, runner.py files in the AML root directory. Can these files read the corresponding files like cp2k.py?  Thanks in advance for your help.

Sincerely,
Zhiqiang

**Comment by cschran** [comment]
Dear Zhiqiang,

Thanks for your interest in our code. You can read in cp2k data with `aml`. The example [from-trajectroy-to-model](https://github.com/MarsalekGroup/aml/tree/main/examples/from-trajectroy-to-model) actually contains all relevant code:

```
fn_positions = dir_trj / 'zundel-pos-1.xyz'
fn_forces = dir_trj / 'zundel-frc-1.xyz'

frames = aml.read_frames_cp2k(fn_positions=fn_positions, fn_forces=fn_forces)
structures = aml.Structures.from_frames(frames, stride=stride_trj, probability=1.0)
```

Note that you might have to also add cell information (in atomic units) to `read_frames_cp2k`, if you deal with perodic systems.

All the best,
Christoph

**Reply by lizhiqiang778** [reply]
Thanks for your reply. My inital idea is that if I have a training set of N2P2, and I want to use aml to screen those representative structures😀. I am not sure this idea is reasonable and reliable for aml. What do you think of this idea?

**Reply by cschran** [reply, marked as answer]
if you have existing training data that you want to prune, you can simply read it in via 
```
structures = aml.Structures.from_file('input.data')
```
and use that for the QbC process. Just make sure that you're setting `n_candidate` to something that is smaller than the training set size.

**Reply by lizhiqiang778** [reply]
Great thanks for your instruction， I can't wait to try it😄.
